app/controllers/barista.rb

```diff
@@ -0,0 +1,25 @@
+require 'orders/use_case/get_orders_by_status'
+require 'orders/use_case/update_order_status'
+require 'order_status'
+
+class Barista
+  attr_accessor :currently_processing_order
+
+  def initialize(order_gateway)
+    @order_gateway = order_gateway
+  end
+
+  def begin_processing_next_order
+    @currently_processing_order = GetOrdersByStatus.new(GetOrdersByStatus::Request(OrderStatus::PROCESSING, 1))
```

Comment:
Moving constructional logic out of controllers will help making dependency changes.

If you look at the projects I shared, you will see a use case factory.